python/cudaq/kernel/ast_bridge.py

```diff
@@ -1127,6 +1128,33 @@ def visit_Attribute(self, node):
         if isinstance(node.value,
                       ast.Name) and node.value.id in self.symbolTable:
             value = self.symbolTable[node.value.id]
+            if cc.PointerType.isinstance(value.type):
+                eleType = cc.PointerType.getElementType(value.type)
+                if cc.StructType.isinstance(eleType):
+                    # Handle the case where we have a struct member extraction
+                    memberName = node.attr
+                    structName = cc.StructType.getName(eleType)
+                    structIdx = None
+                    _, userType = globalRegisteredTypes[structName]
+                    for i, (k, _) in enumerate(userType.items()):
+                        if k == memberName:
+                            structIdx = i
+                    if structIdx == None:
+                        self.emitFatalError(
+                            f'Invalid struct member: {structName}.{memberName} (members={[k for k,_ in userType.items()]})'
```

Comment:
Does this happen? Can a negative test generate this error?

Reply:
Well since the kernel body is just treated as a string that is parsed to an AST, technically one could put anything in there. This is the thing I'm trying to throw an error on 
```python
class TestClass:
    i : int 

@cudaq.kernel 
def test(input : TestClass):
   localVar = input.thisMemberVarDoesNotExist
```

I added a negative test for it.